src/dmd/parse.d

```diff
@@ -3402,7 +3402,19 @@ final class Parser(AST) : Lexer
 
         case TOK.int64:
             t = AST.Type.tint64;
-            goto LabelX;
+            nextToken();
+            if (token.value == TOK.int64)   // if `long long`
+            {
+                error("use `long` for a 64 bit integer instead of `long long`");
+                nextToken();
+            }
+            else if (token.value == TOK.float64)   // if `long double`
+            {
+                error("use `real` for an 80 bit floating point type instead of `long double`");
```

Comment:
Or word it in such a way that omits the type size.

Reply:
> (Note that this also concerns DMD/VC++ on Win64.)

DMD has 80-bit floats with VC, too, but LDC does not.

> Or word it in such a way that omits the type size.

I think this would be best, as it avoids a dependency on the backend.